pkg/operators/oci/oci.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocioperator
```

Comment:
In general, I think the OCI package and this OCI operator need to have a clearer difference. IMO, the OCI package shouldn't have any IG-specific thing but simply helpers like `GetManifestForHost`, `GetContentFromDescriptor`, `EnsureImage`, etc. Instead, the OCI operator should be the wrapper on top of it implementing `GetMetadataFromManifest` or things like that. However, I understand it won't be possible until the image command is not using this refactoring and we remove the run implementation as a gadget.

Is it also your idea?